src/components/Documents/DocumentCard.jsx

```diff
@@ -0,0 +1,222 @@
+ * @property {string} fileUrl - The file URL of the document
+ */
+
+/**
+ * DocumentCard - A component that displays a single document's information
+ * and provides actions (preview, share, delete) through a context menu.
+ *
+ * @memberof Documents
+ * @name DocumentCard
+ * @param {object} props - Component props
+ * @param {Document} props.document - The document object to be displayed
+ * @param {Function} props.onPreview - Function to handle the "Preview" action
+ * @param {Function} props.onShare - Function to handle the "Share" action
+ * @param {Function} props.onDelete - Function to handle the "Delete" action
+ * @returns {React.JSX.Element} The DocumentCard component
+ */
+const DocumentCard = ({ document, onShare, onDelete, onPreview }) => {
+  const theme = useTheme();
+
+  // State to manage the anchor element for the context menu
+  const [anchorEl, setAnchorEl] = useState(null);
+  // State to control the visibility of the context menu for a specific document
+  const [openMenu, setOpenMenu] = useState(null);
+
+  const handleClick = (event, clickedDocument) => {
+    setAnchorEl(event.currentTarget);
+    setOpenMenu(clickedDocument.id);
+  };
+
+  // Closes the context menu
+  const handleClose = () => {
+    setAnchorEl(null);
+    setOpenMenu(null);
+  };
+
+  // Handles the click on a context menu item, executing the corresponding action
+  const handleMenuItemClick = (action, clickedDocument) => () => {
+    action(clickedDocument);
+    handleClose();
+  };
+
+  // ... (Styling definitions below)
+
+  const rowStyling = {
+    display: 'flex',
+    columnGap: '10px'
+  };
+
+  const columnStyling = {
+    display: 'flex',
+    flexDirection: 'column'
+  };
+
+  const dataStyling = {
+    ...rowStyling,
+    fontSize: '.8rem',
+    justifyContent: 'flex-start'
+  };
+
+  const dataSectionStyling = {
+    ...columnStyling,
+    '& div': {
+      ...dataStyling
+    }
+  };
+
+  const dataColumnStyling = {
+    ...columnStyling,
+    flex: '1 1 90%',
+    rowGap: '10px',
+    '& div': {
+      ...dataSectionStyling
+    }
+  };
+
+  const actionsColumnStyling = {
+    ...columnStyling,
+    flex: '0 1 10%',
+    maxWidth: '54px',
+    justifyContent: 'center',
+    alignItems: 'flex-end'
+  };
+
+  const cardStyling = {
+    ...rowStyling,
+    background: theme.palette.background.tint,
+    justifyContent: 'space-between',
+    gap: '10px'
+  };
+
+  // icon styles
+  const iconSize = {
+    height: '24px',
+    width: '24px'
+  };
+
+  const iconStyling = {
+    width: '100%'
+  };
+
+  return (
+    <Box>
+      <Card sx={{ my: '5px', position: 'relative' }}>
+        <CardContent sx={cardStyling}>
+          <Box sx={dataColumnStyling}>
+            <Box>
+              <Typography
+                sx={{ fontSize: '1rem !important', fontWeight: 'bold' }}
+                variant="body1"
+                component="div"
+                noWrap
+              >
+                {document.name || '[No Name provided]'}
```

Comment:
Would want to truncate files with potentially long names.

![Screenshot 2024-08-27 at 8 12 28 PM](https://github.com/user-attachments/assets/a71cb273-198d-49ec-ab13-b960514cf9f7)

I can no longer access the action buttons because of it.

Reply:
Simple enough, it is done.